R/aab-rstudio-detect.R

```diff
@@ -32,8 +32,17 @@ rstudio <- local({
       args = commandArgs(),
       search = search()
     )
-    d$ver <- if (d$api) asNamespace("rstudioapi")$getVersion()
-    d$desktop <- if (d$api) asNamespace("rstudioapi")$versionInfo()$mode
+
+    if (d$api) {
+      ns <- asNamespace("rstudioapi")
+      d$ver <- if (d$api) ns$getVersion()
+
+      # Pull version from namespace rather than via `utils::packageVersion()`
+      # to avoid slowdown (see https://github.com/r-lib/rlang/pull/1657 and
+      # https://github.com/r-lib/rlang/issues/1422)
+      new_api <- ns[[".__NAMESPACE__."]][["spec"]][["version"]] >= "0.17.0"
```

Comment:
You are comparing version numbers as strings, you probably want `package_version()`? Also, `getNamespaceVersion()` is simpler and better IMO.